tools/convert_feat_to_wholegraph.py

```diff
@@ -16,160 +16,255 @@
-            metadata[feat] = {'shape': list(whole_feat_tensor.shape),
-                              'dtype': str(whole_feat_tensor.dtype)}
-
-            # Round up the integer division to match WholeGraph partitioning scheme
-            subpart_size = -(whole_feat_tensor.shape[0] // -num_parts)
-
-            for part_num in range(num_parts):
-                st = part_num * subpart_size
-                end = (part_num + 1) * subpart_size if part_num != (num_parts - 1) \
-                    else whole_feat_tensor.shape[0]
-                wg_tensor = wgth.create_wholememory_tensor(local_comm, 'continuous', 'cpu',
-                                                          (end - st, *whole_feat_tensor.shape[1:]),
-                                                          whole_feat_tensor.dtype, None)
-                local_tensor, _ = wg_tensor.get_local_tensor(host_view=True)
-                local_tensor.copy_(whole_feat_tensor[st:end])
-                filename = wgth.utils.get_part_file_name(feat.replace('/', '~'),
-                                                         part_num, num_parts)
-                wg_tensor.local_to_file(os.path.join(wg_folder, filename))
-                wgth.destroy_wholememory_tensor(wg_tensor)
-
-            # Delete processed feature from memory
-            for t in feats_data:
-                del t[feat]
+    feats_data = []
+
+    # When 'use_low_mem' is not enabled, this code loads and appends different features from
+    # individual partitions. Then each feature is converted into the WholeGraph format.
+    if not use_low_mem:
+        # Read features from file
+        for path in (os.path.join(folder, name) for name in part_files):
+            feats_data.append(dgl.data.utils.load_tensors(f"{path}/{file_name}"))
+        num_parts = len(feats_data)
+        for type_name, feats in fname_dict.items():
+            for feat in feats:
+                feat = type_name + "/" + feat
+                if feat not in feats_data[0]:
+                    raise RuntimeError(
+                        f"Error: Unknown feature '{feat}'. Files contain \
+                                       the following features: {feats_data[0].keys()}."
+                    )
+                logging.info(f"Processing '{feat}' features...")
+                whole_feat_tensor = torch.concat(
+                    tuple(t[feat] for t in feats_data), dim=0
+                )
+                # Delete processed feature from memory
+                for t in feats_data:
+                    del t[feat]
+                wholegraph_processing(
+                    whole_feat_tensor, metadata, local_comm, feat, wg_folder, num_parts
+                )
+
+    # This version is less memory-consuming. For each feature, it iterates through all the
+    # partitions, loading the available features, but only stores (concatenating) only the
+    # current feature to be converted to WholeGraph.
```

Comment:
Can you also provide information of the minimum memory requirement for low-mem and non-low-mem?
I think the minimum memory requirement for low-men is 2X the largest node/edge feature of a graph.

Reply:
Added.